src/app/core/ui/routed-view/routed-view.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component } from "@angular/core";
+import { CommonModule } from "@angular/common";
+import { RouteTarget } from "../../../app.routing";
+import { ActivatedRoute } from "@angular/router";
+import { DynamicComponentDirective } from "../../config/dynamic-components/dynamic-component.directive";
+import { ViewConfig } from "../../config/dynamic-routing/view-config.interface";
+
+/**
+ * Wrapper component for a primary, full page view
+ * that takes parameters from the route and passes these on to normal @Input properties.
+ *
+ * This allows to develop functional feature components in a way to easily reuse them for display
+ * as a full page view or in a modal dialog.
+ */
+@RouteTarget("RoutedView")
+@Component({
+  selector: "app-routed-view",
+  standalone: true,
+  imports: [CommonModule, DynamicComponentDirective],
+  template: `<ng-container
+    *ngIf="component"
+    [appDynamicComponent]="{ component: component, config: config }"
+  ></ng-container>`,
+})
+export class RoutedViewComponent<T = any> {
+  component: string;
+  config: T;
+
+  constructor(route: ActivatedRoute) {
+    route.data.subscribe((data: { component: string } & ViewConfig<T>) => {
+      this.component = data.component;
+      // pass all other config properties to the component as config
+      this.config = Object.assign({}, data.config);
+
+      // merge updated config properties from route params
+      route.paramMap.subscribe((params) => {
+        const config = this.config;
+        for (const key of params.keys) {
+          config[key] = params.get(key);
+        }
+        this.config = { ...config };
+      });
+
+      // TODO: should we also assigned queryParams from the url to input properties?
```

Comment:
I am not sure whether this could result in a problem at one point. There are no restrictions as to what is written into the params so it would be possible to overwrite other config options from there by using the same key. Don't you think the query params object on the route object is already easy enough to use?

Reply:
probably true. I also had some concerns about potential security issues.